broadinstitute/single_cell_portal_core
# Suggest sync when large file is selected for upload (SCP-4013)

This suggests using sync upon selecting a >= 1 GiB for upload.  It improves discoverability of a way to add data that's sometimes much faster than SCP's default.

Syncing is often faster than the SCP upload UI for large files in these cases, ordered by descending fitness:
1. Files are already in Terra workspace
2. Files are already in another Google bucket, and user is comfortable on command line
3. Files are not yet in bucket, and user has a very fast upload connection, is comfortable on command line, and sometimes mistakenly navigates away from upload page or closes browser tab while uploading

This also links to relevant docs from atop the sync UI.

Here's how it looks:

https://user-images.githubusercontent.com/1334561/156242169-40124298-0c8c-48d7-bacd-83be69f371f4.mov

For now, this suggestion is limited to users who are registered for Terra.  Refining the UX and docs for Terra-unregistered users is tracked in another ticket.  Documentation refinements to emphasize copying from another bucket (instead of uploading from a local machine) are also tracked separately.

This also refactors a fair amount of CSFV code to improve cohesion and remove obsolete code.  Previously, very similar code to start validating local and remote files was spread over three files.  Now it's in one file, `validate-file.js`.  Code for CSFV logging is now also centralized, separately, in `log-validation.js`.  This reduces duplication, helps maintainability, and eases future refinements and consolidations.

To test:
* Go to upload UI
* Select a file that is > 1 GiB in size
* Confirm sync suggestion appears
* Click suggested link
* Confirm it opens the sync UI in another tab
* Click "What is sync?" and "Adding data with gsutil" links
* Confirm they open related docs in another tab

This satisfies SCP-4013.
